### PR TITLE
fix votekicks being always unavailable unless ghosted

### PR DESCRIPTION
**Pull request by MilonPL**
## About the PR

Removed the `IsGhost` check and the associated dependencies in the `VotingSystem` class.

## Why / Balance

The `IsGhost` check was redundant because the server already performs eligibility checks for votekick initiators, based on the configuration and player status. It created an issue where, despite having the `votekick.initiator_ghost_requirement` CVar set to false, you still couldn't call a votekick if not ghosted.

## Technical details

Deleted the `IsGhost` check in the `OnVotePlayerListResponseEvent` method, allowing the event to be invoked for all users who passed the server-side checks. Removed the `GhostSystem` dependency and the `_ghostSystem` field from `VotingSystem`.

## Requirements

- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**

no cl, no fun

### CODE REVIEW

**Comment by Lyndomen** [comment]
thanks gamer, periapsis sings your name

File: Content.Client/Voting/VotingSystem.cs [hunk 1]
```diff
@@ -8,8 +7,6 @@ public sealed class VotingSystem : EntitySystem
 
     public event Action<VotePlayerListResponseEvent>? VotePlayerListResponse; //Provides a list of players elligble for vote actions
 
-    [Dependency] private readonly GhostSystem _ghostSystem = default!;
-
```
Comment (metalgearsloth) [inline Content.Client/Voting/VotingSystem.cs:12]: Wouldn't you be better off checking the cvar on client then?

Comment (MilonPL) [reply Content.Client/Voting/VotingSystem.cs:12]: All of votekick cvars are server only and the event checks for eligiblity on the server already, so I don't think there's much point in also doing it on the client now

**Comment by deathride58** [comment]
This PR's functionality is superceded by #32528 and can be closed.

**Pull request closed without merging**